sys/kern/ringbuf.c

```diff
@@ -28,6 +28,24 @@ static void consume(ringbuf_t *buf, unsigned bytes) {
     buf->tail = 0;
 }
 
+/* below two function are to help track ringbuf pointers in case when it is
+ * changed by something else. e.q. DMA */
+void ringbuf_produce(ringbuf_t *buf, unsigned bytes) {
+  buf->count += bytes;
+  buf->head += bytes;
+  buf->head %= buf->size;
+  if (buf->count > buf->size)
+    buf->count = buf->size;
```

Comment:
If the buf->size is quite big, the buf->count can overflow and we will not notice it.

Reply:
In case of DMA and my usage of this function it is not a problem. I will add assertion.